src/test/java/com/github/sdonkov/PrefixTreeTest.java

```diff
@@ -1,28 +1,76 @@
 package com.github.sdonkov;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jol.info.GraphLayout;
 
 
 import java.io.IOException;
 import java.net.URISyntaxException;
 import java.nio.file.Files;
 import java.nio.file.Paths;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.TimeUnit;
 
 import static org.junit.jupiter.api.Assertions.*;
 
-
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@Fork(value = 1, warmups = 1)
+@State(Scope.Benchmark)
 public class PrefixTreeTest {
 
     private static List<String> words;
     private static PrefixTree prefixTree;
+    private final static Logger LOGGER = LogManager.getLogger(PrefixTree.class);
 
     @BeforeAll
     static void setUp() throws IOException, URISyntaxException {
         prefixTree = new PrefixTree();
-        words = Files.readAllLines(Paths.get(PrefixTreeTest.class.getResource("/tests.txt").toURI()));
+        words = Files.readAllLines(Paths.get(PrefixTreeTest.class.getResource("/dictionary.txt").toURI()));
+        words.forEach(word -> prefixTree.add(word));
+    }
+
+    @Test
```

Comment:
We definitely do not want to run this along with the unit tests.
It should be a separate command that triggers perf testing so that development is not hindered. While changing the code, the dev often needs a quick check whether something is broken. The perf tests are something to run sparingly, e.g. prior to posting a pull request that has the potential to regress or improve on the performance front.

It does make sense to move the perf testing to a separate file. Files are useful to avoid collisions during development. Ideally, you should have a single reason to change a file. This file contains tests for the API contract of the PrefixTree. Changing the contract should be the only reason to touch it.